src/server/endpoints/covidcast.py

```diff
@@ -132,22 +92,29 @@ def handle():
     fields_float = ["value", "stderr", "sample_size"]
     is_compatibility = is_compatibility_mode()
     if is_compatibility:
-        q.set_order("signal", "time_value", "geo_value", "issue")
+        q.set_sort_order("signal", "time_value", "geo_value", "issue")
     else:
         # transfer also the new detail columns
         fields_string.extend(["source", "geo_type", "time_type"])
-        q.set_order("source", "signal", "time_type", "time_value", "geo_type", "geo_value", "issue")
-    q.set_fields(fields_string, fields_int, fields_float)
+        q.set_sort_order("source", "signal", "time_type", "time_value", "geo_type", "geo_value", "issue")
+    q.set_select_fields(fields_string, fields_int, fields_float)
 
     # basic query info
     # data type of each field
     # build the source, signal, time, and location (type and id) filters
 
-    q.where_source_signal_pairs("source", "signal", source_signal_pairs)
-    q.where_geo_pairs("geo_type", "geo_value", geo_pairs)
-    q.where_time_pair("time_type", "time_value", time_pair)
+    q.apply_source_signal_filters("source", "signal", source_signal_filters)
+    q.apply_geo_filters("geo_type", "geo_value", geo_filters)
+    q.apply_time_filter("time_type", "time_value", time_filter)
 
-    _handle_lag_issues_as_of(q, issues, lag, as_of)
+    # Handle history table in this priority order: issues > lag > as_of
+    # if all are None, we are using the (standard/default) `latest_table`, to fetch the most recent issue quickly
+    if issues:
+        _apply_issues_filter(q, issues)
+    elif lag is not None:
+        _apply_lag_filter(q, lag)
+    elif as_of is not None:
+        _apply_as_of_filter(q, as_of)
```

Comment:
it seems silly to check `if field` / `if field is not None` here when each of these methods do the same test internally.  it also seems silly to silently leave out a filtering condition for `lag` when `issue` is also specified, or to leave out `as_of` when `lag` or `issue` is specified...  i claim that either (1) these 3 conditions should all be applied to the resulting SQL if they are specified in any combination, or (2) an error should be produced if they are truly intended to be mutually exclusive.  i think it is more correct to do the former; that is, leave out the `if/elif/elif` lines and just call the 3 methods here.  thoughts?

Reply:
Applying all three makes sense if it won't break any existing workflows!